### PR TITLE
Python: Enable mypy for the HuggingFace connectors. Increase unit test code coverage.

### LINKED ISSUE #7133: Python: CQ Spike: HuggingFace
Remove:

```
[mypy-semantic_kernel.connectors.ai.hugging_face.*]
ignore_errors = true
```
from mypy.ini

And improve ai.hugging_face.* test coverage to over 95% without `pragma: no cover`

### PR DESCRIPTION
**Pull request by moonbox3**
### Motivation and Context

We have mypy enabled on parts of the code base, but not all. The goal is to enable it across the entire SK python code. As part of this, we've broken up the work to tackle different sections. Additionally, we're working to improve the unit test code coverage for these sections of code.

### Description

This PR:
- turns on mypy for the HuggingFace connector
- adds more unit test coverage for the text completion and embeddings connector to achieve >95% code coverage.
- closes #7133

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/bdf7f39515157e70289d85eb68a3c88c47c92c90/README.md">Py3.10 Test Coverage</a><details><summary>Python 3.10 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>semantic_kernel/connectors/ai/hugging_face/services</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/bdf7f39515157e70289d85eb68a3c88c47c92c90/python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py">hf_text_completion.py</a></td><td>43</td><td>14</td><td>67%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/bdf7f39515157e70289d85eb68a3c88c47c92c90/python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py#L92-L93">92&ndash;93</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/bdf7f39515157e70289d85eb68a3c88c47c92c90/python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py#L95">95</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/bdf7f39515157e70289d85eb68a3c88c47c92c90/python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py#L121-L144">121&ndash;144</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/bdf7f39515157e70289d85eb68a3c88c47c92c90/python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_embedding.py">hf_text_embedding.py</a></td><td>28</td><td>9</td><td>68%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/bdf7f39515157e70289d85eb68a3c88c47c92c90/python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_embedding.py#L8">8</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/bdf7f39515157e70289d85eb68a3c88c47c92c90/python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_embedding.py#L44-L45">44&ndash;45</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/bdf7f39515157e70289d85eb68a3c88c47c92c90/python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_embedding.py#L54-L59">54&ndash;59</a></td></tr><tr><td><b>TOTAL</b></td><td><b>6804</b></td><td><b>624</b></td><td><b>91%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python 3.10 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1686 | 1 :zzz: | 0 :x: | 0 :fire: | 25.224s :stopwatch: |

File: python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
 
 import logging
+import sys
 from collections.abc import AsyncGenerator
 from threading import Thread
-from typing import TYPE_CHECKING, Any, Literal
+from typing import Any, Literal
+
+if sys.version_info >= (3, 12):
+    from typing import override
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py:10]: Add pragma here as well!

File: python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py [hunk 2]
```diff
@@ -88,9 +92,12 @@ async def get_text_contents(
             List[TextContent]: A list of TextContent objects representing the response(s) from the LLM.
         """
         try:
+            if not isinstance(settings, HuggingFacePromptExecutionSettings):
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py:95]: This shouldn't be in the try block

**Review by juliomenendez — APPROVED** [review]

**Review by eavanvalkenburg — APPROVED** [review]
Some nits, overall looks good!

**Review by juliomenendez — APPROVED** [review]

**Pull request merged**